flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/descriptors/ConnectTableDescriptor.java

```diff
@@ -80,51 +87,23 @@ public void registerTableSink(String name) {
 	 *
 	 * @param name table name to be registered in the table environment
 	 */
-	@Override
 	public void registerTableSourceAndSink(String name) {
 		registerTableSource(name);
 		registerTableSink(name);
 	}
 
-	/**
-	 * Specifies the format that defines how to read data from a connector.
-	 */
-	@Override
-	public D withFormat(FormatDescriptor format) {
-		formatDescriptor = Optional.of(Preconditions.checkNotNull(format));
-		return (D) this;
-	}
-
-	/**
-	 * Specifies the resulting table schema.
-	 */
-	@Override
-	public D withSchema(Schema schema) {
-		schemaDescriptor = Optional.of(Preconditions.checkNotNull(schema));
-		return (D) this;
-	}
-
 	/**
 	 * Converts this descriptor into a set of properties.
 	 */
 	@Override
 	public Map<String, String> toProperties() {
```

Comment:
How about we make this method a bit safer to extend. Right now it adds implicit contract that you have to call `super.toProperties` in a child class.

Maybe we could change it to:
```
public final Map<String, String> toProperties() {
  ...
  properties.asMap().putAll(extraProperties());
}

protected abstract Map<String, String> extraProperties();
```

Reply:
I just saw it's done this way already for `FormatDescriptor`